### ISSUE TITLE
New Java versions lead to FN about the rule DMI_INVOKING_TOSTRING_ON_ARRAY

### ISSUE THREAD

**Issue by Zustin** [spotbugs/spotbugs#1874]
Hi, I found a false negative about the rule [DMI_INVOKING_TOSTRING_ON_ARRAY](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html#useless-string-invocation-of-tostring-on-an-array-dmi-invoking-tostring-on-array). This case is found by using SpotBugs CLI and IDE, but class files are generated by different Java versions. Please refer to the following case, the class file generated by `java-8` can be detected, but class files generated by `java-11,16,17` cannot be detected.

```java
public class Test {
    private static final String[] gargs = new String[] { "1", "2" };
    public void print() {
        System.out.println("" + gargs);  // should report a warning here
    }
}
```
**Expected Outcome:** A warning at line 4, [Outcomes Repo](https://github.com/Zustin/Issue1874)
SpotBugs version: 4.5.2 and run by CLI
